### PR TITLE
The detomatix cart now sends forged messages that make PDAs explode if one tries to reply to them instead of detonating them right off the bat.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Exactly what it's written on the tin. I have reworked the detomatix cartridge after reading the detomatix removal PR again (including comments and suggestions). The forged name and job of the message respect filters to avoid bad words, the message uses the send_message proc (it has a few new args) and the PDA shouldn't explode outside replying to a forged bomb message (or opening the message menu within 10 seconds after receiving it). I have also changed the TC cost of the item to 4 and the number of charges to 6 considering that, even in the hands of a cunning player, not all messages will be "replied". This buff to the tc cost and charges was done with the assumption that half of these messages will be ignored anyway in general. 

## Why It's Good For The Game
The current way detomatix works gives no room to the receivers to avoid getting pwned. A few people say it's uninteresting and not a sufficient source of interaction or amusement amongst players (though my opinions on this may diverge a bit) and that the explosion is too strong (alas, this is more a general issue with explosion RNG and thus should be handled in a separate PR). On the other side I have heard people arguing it adds value to the game in the form of paranoia and sabotage. I have also heard there's a plan to replace PDAs with modular computer tablets, though it's too early in the works to warrant a removal of this cartridge, especially when half of the station crew doesn't start with modular tablets and there's no exact counterpact to PDA messaging on modular computers (the closest is a chat app).

This PR resolves the issue of detomatix being unavoidable without the ultimatum of a removal that most people don't want anyway. Basically, an alternative to #62484. And why yes, I have tested it.

## Changelog

:cl: Ghommie (Thanks to dorsisdwarf for the idea)
balance: The detomatix cart now sends forged messages (name and job of the message can also be forged) that can explode a PDA when someone tries to reply to them or access the message menu of the PDA UI up to 10 seconds after receiving it instead of detonating the PDA right off the bat. Because of this and the possibility that a portion of these messages goes "unreplied", the TC cost of the item has also been reduced to 4 and the number of charges upped to 6.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
The window should be longer than 10 seconds, maybe even slightly randomized to prevent people from gaming it.

**Comment by zxaber** [comment]
Does the message show up in the message logs on the PDA?

Also, I'd probably just remove the limit. There's no real functional reason it needs to exist (assuming it respects adjacent checks). If someone wants to wait half an hour and *then* reply to a trapped message, let them blow up all the same.

**Comment by itseasytosee** [comment]
There is hot **single** _space_ women on your station!! **To find out click here**!!! ---> ---> (**reply**)

**Comment by iwishforducks** [comment]
Funny but I don't like the idea of powergaming this introduces and not replying to PDA messages. I'd rather be tormented with my PDA randomly exploding rather than punishing me for actually using its functions.

**Comment by Ghommie** [comment]
> 
> Does the message show up in the message logs on the PDA?
> 
Yes.

> Also, I'd probably just remove the limit. There's no real functional reason it needs to exist (assuming it respects adjacent checks). If someone wants to wait half an hour and _then_ reply to a trapped message, let them blow up all the same.

Oh turns out I didn't word it out very well. The time window is exclusive to opening the message menu in case of smartasses using the message menu to reply to the message rather than the reply button (See https://github.com/tgstation/tgstation/pull/62484#issuecomment-955790115). Time doesn't matter for the href buttons.

> 
> 
> The window should be longer than 10 seconds, maybe even slightly randomized to prevent people from gaming it.

See above. I can still extend and randomize it out if you want.

**Comment by phasenoisepon** [comment]
> I don't get what you're referring to with powergaming by the by.

*beep* Message from `JOHN ANDERS: CHEF` reply?

[Common] Valid McSalad: Hey John, did you just PDA me?
[Common] John Anders: No?
[Common] Valid McSalad: AI LAW 2 TURN THE MESSAGING SERVER OFF WE HAVE TRAITORS

**Comment by Ghommie** [comment]
That's quite an exaggerated example considering message server consoles can too send messages with sham sender name and job and that disabling the entire messaging server in spite of a single anomalous message is quite the bold move.

But thanks for pointing it out. I'll work on a way get the pda bomb messages to be rerouted to the hub if the message server is down and then broadcasted anyway. The only thing the message server does, outside of relaying request console and pda messages, is store them so they can be later checked with the message server comp.

**Comment by ATH1909** [comment]
> (or opening the message menu within 10 seconds after receiving it)

instead of this, just wait until they either reply to the message OR send a direct message to the person who sent them that rigged message. in other words, pda messaging the bomber in any way at any later time should cause the target's pda to explode.

**Comment by Ghommie** [comment]
> instead of this, just wait until they either reply to the message OR send a direct message to the person who sent them that rigged message. in other words, pda messaging the bomber in any way at any later time should cause the target's pda to explode.

Since name and job of the message signal can be forged, someone may as well send a pda message to someone with an alias that doesn't show up in the PDA messaging menu, so doing that may require adding a list variable to the PDA type that injects the list of aliases used to message PDA bombs and whatsnot in the PDA list of the message menu.
I'll give it a shot though.

**Comment by itseasytosee** [comment]
We could turn them into emp bombs that also remove a random few access levels from an inserted card. Although that nerf alongside this one would be too much.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -224,6 +224,10 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(user, span_warning("You don't have the dexterity to do this!"))
 		return
 
+	if(HAS_TRAIT(src, TRAIT_PDA_MESSAGE_MENU_RIGGED) && mode == 2)
```
Comment (Kylerace) [inline code/game/objects/items/devices/PDA/PDA.dm:227]: what the fuck is 2

Comment (Ghommie) [reply code/game/objects/items/devices/PDA/PDA.dm:227]: One of the many magic numbers for the mode of the pda UI. I'll fix em soon.

**Comment by Kylerace** [comment]
holy fuck those magic numbers thank you for cleaning those up

**Comment by Kylerace** [comment]
> 
> 
> > I don't get what you're referring to with powergaming by the by.
> 
> _beep_ Message from `JOHN ANDERS: CHEF` reply?
> 
> [Common] Valid McSalad: Hey John, did you just PDA me? [Common] John Anders: No? [Common] Valid McSalad: AI LAW 2 TURN THE MESSAGING SERVER OFF WE HAVE TRAITORS

this could also be comms agents

**Comment by PolishUser** [comment]
Cool, you've moved The detomatix cartridge item into obscurity. No one in the right mind will buy this item because it is dependent on the recipient of the message. Normal people keeping that in mind will wait out those magical 10 seconds and then reply or even better, don't respond to them at all. In addition, I won't be surprised if some maintainer No. 563 will swoop in later on and just remove this antag item completely. Anyways, why tg didn't get rid of PDA's completely? Tablets are the future.

**Pull request merged**